### ISSUE TITLE
Support behat --rerun

### ISSUE THREAD

**Issue by dkarlovi** [jakzal/phpqa#43]
Behat's `--rerun` stores failed tests into `/tmp` by default. Since the container is destroyed after each run, that data is also purged. It might make sense to support mounting a volume / scratch disk for the image which would allow supporting this feature.

**Comment by dkarlovi** [comment]
That's my suggestion, supporting binding local `/tmp` into container's `/tmp`?

**Comment by jakzal** [comment]
You can already pass `-v /tmp/phpqa:/tmp` while running the container, no?

**Comment by dkarlovi** [comment]
Yes, but that means the user needs to know Behat internals for this to work (which is what I needed to dig through to figure out why it didn't).

Maybe a way to mark up pre-requisites and a custom entrypoint which would validate them?

```json
    {
      "name": "behat",
      "summary": "Helps to test business expectations",
      "website": "http://behat.org/",
      "command": {
        "box-build": {
          "repository": "https://github.com/behat/behat.git",
          "phar": "behat.phar",
          "bin": "/usr/local/bin/behat"
        }
      },
      "prereqs": [
          {"type": "tmp"}
      ],
      "test": "behat --version"
    },
```

Or something similar?

**Comment by jakzal** [comment]
If we decide to do it, I'd just update Dockerfiles with volume definitions. I need to think if it's the right thing to do though (when I'm back from holidays ;) ). I was very careful to only configure a bare minimum for each tool.

**Comment by jakzal** [comment]
Ok, I think it would make sense to define a `VOLUME /tmp`, since this is the tmp location for most of the tools (as `sys_get_temp_dir()` points there).

@dkarlovi Would you like to send a PR?

**Comment by dkarlovi** [comment]
Yes, so the gist would be how to best explain what's need doing and verify it's done. Declaring `VOLUME` would create a temporary volume is a named one is not declared, it would defeat the purpose.
